cmd/eksctl-anywhere/cmd/upgradecluster.go

```diff
@@ -114,10 +133,32 @@ func (uc *upgradeClusterOptions) upgradeCluster(cmd *cobra.Command) error {
 		return err
 	}
 
+	_, err = time.ParseDuration(uc.cpWaitTimeout)
+	if err != nil {
+		logger.V(0).Info(fmt.Sprintf("Warning: Invalid %s value: %s, Using the default timeout instead: %s",
+			cpWaitTimeoutFlag, uc.cpWaitTimeout, clustermanager.DefaultControlPlaneWaitStr))
+		uc.cpWaitTimeout = clustermanager.DefaultControlPlaneWaitStr
+	}
+
+	_, err = time.ParseDuration(uc.externalEtcdWaitTimeout)
+	if err != nil {
+		logger.V(0).Info(fmt.Sprintf("Warning: Invalid %s value: %s, Using the default timeout instead: %s",
+			externalEtcdWaitTimeoutFlag, uc.externalEtcdWaitTimeout, clustermanager.DefaultEtcdWaitStr))
+		uc.externalEtcdWaitTimeout = clustermanager.DefaultEtcdWaitStr
+	}
+
+	var perMachineWaitTimeout time.Duration
+	perMachineWaitTimeout, err = time.ParseDuration(uc.perMachineWaitTimeout)
```

Comment:
nit: you don't need to declare the var separately
```suggestion
	perMachineWaitTimeout, err := time.ParseDuration(uc.perMachineWaitTimeout)
```